src/main/java/ch/njol/skript/expressions/ExprVectorLength.java

```diff
@@ -61,43 +58,44 @@ public Class<?>[] acceptChange(ChangeMode mode) {
 	}
 
 	@Override
-	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
 		assert delta != null;
-		final Vector v = getExpr().getSingle(e);
+		final Vector v = getExpr().getSingle(event);
 		if (v == null)
 			return;
-		double n = ((Number) delta[0]).doubleValue();
+		double deltaLength = ((Number) delta[0]).doubleValue();
 		switch (mode) {
+			case REMOVE:
+				deltaLength = -deltaLength;
+				//$FALL-THROUGH$
 			case ADD:
-				if (n < 0 && v.lengthSquared() < n * n) {
+				if (deltaLength < 0 && v.lengthSquared() < deltaLength * deltaLength) {
 					v.zero();
 				} else {
-					double l = n + v.length();
+					double l = deltaLength + v.length();
 					v.normalize().multiply(l);
```

Comment:
You might want to check whether the vector is already normalised -- bucket may do this already though I don't remember.